confetti/src/main/java/com/github/jinatonic/confetti/confetto/Confetto.java

```diff
@@ -281,13 +278,23 @@ public boolean applyUpdate(long passedTime) {
         startedAnimation = animatedTime >= 0;
 
         if (startedAnimation && !terminated) {
-            currentX = computeDistance(animatedTime, initialX, initialVelocityX, accelerationX,
+            float[] pairs = new float[2];
```

Comment:
move this into instance variable to avoid allocating this array on every draw, call it `workPairs` or something.